### ISSUE TITLE
cmd/go: malformed import path error for paths with plus sign in go1.16 

### ISSUE THREAD

**Issue by siadat** [golang/go#45158]
Go 1.15 used to compile package names with + signs in them, however, in Go 1.16 I get an error (tested on both macOS and Linux).

### What version of Go are you using?

<pre>
$ go version # darwin
go version go1.16.2 darwin/amd64

$ go version # linux
go version go1.16.2 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

Let's create 3 files:

```
// go.mod
module github.com/siadat/a+b
go 1.16
```

```
// x/x.go
package x
type X int
```

```
// y/y.go
package y
import "github.com/siadat/a+b/x"
type X y.X
```

Then
``` go build ./...```

### What did you expect to see?

No errors

### What did you see instead?

```
malformed import path "github.com/siadat/a+b/x": invalid char '+'
malformed import path "github.com/siadat/a+b/y": invalid char '+'
```

**Comment by AlexRouSg** [comment]
Duplicate of #44776
Backport issue for 1.16 is #44885

You can try with tip to see if the patch works.

**Comment by jayconrod** [comment]
Closing as duplicate, per @AlexRouSg's comment. This should be fixed in 1.16.3.

`+` is not allowed in module paths, but the main module and replaced modules may break some of the rules, which I think is the case here.
